### PR TITLE
MAINT: Fix `-Wunused-result` warnings in `sparse/linalg/dsolve`

### PR DESCRIPTION
**Pull request by Smit-create**
#### Reference issue

* References: https://github.com/rgommers/scipy/issues/42, https://github.com/scipy/scipy/pull/14442#issuecomment-890372841

#### Additional information

cc @rgommers

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @Smit-create. That's a lot of messy code unfortunately. xref my comment in https://github.com/rgommers/scipy/pull/56#discussion_r691009525 about actually checking that the returned values are `0`/`NULL` - you said there you were still going to do that?

**Comment by Smit-create** [comment]
> you said there you were still going to do that?

Yes, will add check for `(fscanf(...) == 0)` too.

**Comment by Smit-create** [comment]
Please have a look at this. @rgommers

File: scipy/sparse/linalg/dsolve/SuperLU/SRC/creadhb.c [hunk 1]
```diff
@@ -298,42 +305,63 @@ creadhb(FILE *fp, int *nrow, int *ncol, int *nonz,
...
 #endif
 
     /* Line 2 */
     for (i=0; i<5; i++) {
-	fscanf(fp, "%14c", buf); buf[14] = 0;
-	sscanf(buf, "%d", &tmp);
+	f_count = fscanf(fp, "%14c", buf); buf[14] = 0;
+  check_read(f_count);
+	s_count = sscanf(buf, "%d", &tmp);
+  check_read(s_count);
```
Comment (rgommers) [inline scipy/sparse/linalg/dsolve/SuperLU/SRC/creadhb.c:332]: mixed tabs and spaces here as well

File: scipy/sparse/linalg/dsolve/SuperLU/SRC/creadhb.c [hunk 2]
```diff
@@ -157,16 +160,20 @@ static int ReadVector(FILE *fp, int n, int *where, int perline, int persize)
...
     register float realpart;
-    char tmp, buf[100];
-    
+    char tmp, buf[100], *dummy;
+
     i = pair = 0;
     while (i < n) {
-	fgets(buf, 100, fp);    /* read a line at a time */
+	dummy = fgets(buf, 100, fp);    /* read a line at a time */
+  if(dummy == NULL) {
```
Comment (rgommers) [inline scipy/sparse/linalg/dsolve/SuperLU/SRC/creadhb.c:174]: there's something funky going on here with mixed tabs and spaces

**Review by rgommers — COMMENTED** [review]
There is something deeply messed up with these source files, there's lots of mixed tabs-spaces which messes with formatting. A `:retab` in Vim gives large diffs. It doesn't seem like a good idea to fix that here. 

Most of this diff is clean, there's just a couple of places where the indentation is odd because there were tabs already there. But that is what it is, let's not mess with that more.

**Comment by rgommers** [comment]
Merged, thanks @Smit-create!

It would be good to upstream these changes to https://github.com/xiaoyeli/superlu, without the whitespace changes. Could you try that?

**Comment by Smit-create** [comment]
Yes, thanks!

**Pull request merged**